lib/events.js

```diff
@@ -167,10 +167,16 @@ MapboxEventManager.prototype = {
       proximity: proximity,
       limit: geocoder.options.limit,
       // routing: search.routing, //todo --> add to plugin
-      queryString: (geocoder._inputEl) ? geocoder._inputEl.value : geocoder.inputString,
       mapZoom: zoom,
       keyboardLocale: this.locale
     }
+
+    if (event === "search.select"){
+      payload.queryString = geocoder.inputString
+    }else{
+      payload.queryString = (geocoder._inputEl) ? geocoder._inputEl.value : geocoder.inputString;
+    }
+    return payload;
```

Comment:
code-style wise, it might be more readable to use an if / else if / else pattern rather than an if statement with a ternary inside one of the cases.